src/services/cartService.js

```diff
@@ -0,0 +1,74 @@
+const fs = require('fs')
+const usersJson = require("../../model/carts/carts.json")
+
+class Cart {
+    constructor(id, timestamp, products) {
+        this.id = id
+        this.timestamp = timestamp
+        this.products = products
+    }
+    getAllCarts = async () => {
+        const getAllCarts = await fs.promises.readFile(`./model/carts/carts.json`, 'utf8')
+        const parsedCarts = JSON.parse(getAllCarts)
+        return parsedCarts
+    }
+    addNewCart = async () => {
+        const getAllCarts = await this.getAllCarts()
+        const newCart = {
+            id: getAllCarts.length + 1,
+            timestamp: new Date().getTime(),
+            products: []
+        }
+        getAllCarts.push(newCart)
+        await fs.promises.writeFile(`./model/carts/carts.json`, JSON.stringify(getAllCarts))
+        return newCart
+    }
+
+    getCartById = async (id) => {
+        const getAllCarts = await this.getAllCarts()
+        const cartFiltered = getAllCarts.filter(cart => cart.id === id)
```

Comment:
metodo correcto para encontrar uno: FIND()

EN TODOS LOS MÉTODOS QUE NECESITEN BUSCAR UNO Y REALIZAR OPERACIONES (MOSTRAR,ACTUALIZAR,ELIMINAR) SE DEBE INCLUIR EL CONDICIONAL "SI NO EXISTE"

Reply:
Perfecto, entiendo. Si uso find() en vez de filter() me devuelve un Object en vez de un Array of Object. 
Para este caso es buena practica devolver solo un {product} en vez de un [{product}]?